models/repo_permission.go

```diff
@@ -23,25 +23,49 @@ func (p *Permission) IsAdmin() bool {
 
 // HasAccess returns true if the current user has at least read access to any unit of this repository
 func (p *Permission) HasAccess() bool {
+	for _, u := range p.Units {
+		if u.AllowAnonymous {
+			return true
```

Comment:
I think `return u.AllowAnonymous` should work here and the if block 
can be dropped

Reply:
I don't think that will be the same and will break logic

Reply:
Whoops, true.

Sorry, I have been writing too much Ruby of late where you 
could ___inline returns with an if block___